akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -439,25 +439,23 @@ final class FoldAsync[In, Out](zero: Out, f: (Out, In) ⇒ Future[Out]) extends
   def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with InHandler with OutHandler {
     val decider = inheritedAttributes.get[SupervisionStrategy].map(_.decider).getOrElse(Supervision.stoppingDecider)
 
-    private var inFlight = 0
-
     private var aggregator: Out = zero
-    private var aggregating: Future[Out] = Future.successful(aggregator)
+    private var aggregated: Promise[Out] = Promise.successful(aggregator)
```

Comment:
Hmm. Why did we need to introduce an additional promise/failure over the one we got from the folding function?